### PR TITLE
You can no longer equip pet collars on humans

### PR DESCRIPTION
**Pull request by CodeJack0711**
:cl: obscolene
tweak: Humans can no longer equip pet collars
/:cl:

### CODE REVIEW

**Comment by adamkad1** [comment]
but we need to put collars on catgrilles

**Comment by MMMiracles** [comment]
it only checks if their a human so catgirls as they are can technically wear it still

**Comment by PKPenguin321** [comment]
that's not what the ishuman check checks for mmmiracles. it just checks for the /human typepath which catgirls, lizards, etc fall under.

either way I'm pretty sure this is the wrong way to do this

**Comment by Qustinnus** [comment]
its not

**Comment by fludd12** [comment]
This most certainly is the wrong way to do this, just remove its slot tag - its not like simple animals have inventory slots anyway.

Also, why?

File: code/modules/clothing/neck/_neck.dm [hunk 1]
```diff
@@ -187,6 +187,11 @@
 	item_color = "petcollar"
 	var/tagname = null
 
+/obj/item/clothing/neck/petcollar/mob_can_equip(mob/M, mob/equipper, slot, disable_warning = 0)
+	if(ishuman(M))
```
Comment (ShizCalev) [inline code/modules/clothing/neck/_neck.dm]:
```
if(ishuman(M))
	var/mob/living/carbon/human/H = M
	if(human.dna?.species?.id == "human")
		return FALSE
```

Comment (Iamgoofball) [reply code/modules/clothing/neck/_neck.dm]: Wait, that's valid syntax now? Oh fuck, that'll let us condense a lot of code.

**Comment by Iamgoofball** [comment]
This is the incorrect fix. Since collars are for pets, they should just be /obj/item/pet_collar. No need for it to be on the clothing subclass to begin with.

File: code/modules/clothing/neck/_neck.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/modules/clothing/neck/_neck.dm]: also no, this is dumb and you know it

**Comment by ShizCalev** [comment]
@Iamgoofball monkeys.

**Comment by ShizCalev** [comment]
No? They're still carbon.

**Comment by FantasticFwoosh** [comment]
Id normally be all for removing it, but people who wear pet collars in the first place **are to be ridiculed anyway** so this is the wrong solution.

**Comment by Dorsisdwarf** [comment]
If we're serious about removing the fetish content can we scrap the schoolgirl and maid outfitstoo?

**Comment by fludd12** [comment]
seriously though why are we going on a witch hunt for code that could distantly reference anything fetish-like.
last i checked space wasn't exactly a puritan society, and i don't want to see the fifty or so joke PRs that will rise from someone's favorite outfit or item being removed.
`cableties removed because citadel uses them a lot`
`electropacks removed because electricity fetish`
`xenobiology removed because slimegirl fetish`
`LiBrArY rEmOvEd BeCaUsE aDuLt SeCtIoN`

**Comment by Iamgoofball** [comment]
I thought bed sheets were backpack slot items? Oh well.

**Comment by obscolene** [comment]
@ShizCalev Is this satisfactory, Mr. Calev?

File: code/modules/clothing/neck/_neck.dm [hunk 2]
```diff
@@ -180,13 +180,19 @@
...
+/obj/item/clothing/neck/petcollar
 	name = "pet collar"
-	desc = "It's for pets. Though you probably could wear it yourself, you'd doubtless be the subject of ridicule."
+	desc = "It's for pets."
 	icon_state = "petcollar"
 	item_color = "petcollar"
 	var/tagname = null
 
+/obj/item/clothing/neck/petcollar/mob_can_equip(mob/M, mob/equipper, slot, disable_warning = 0)
+	if(ishuman(M))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/neck/_neck.dm]:
```suggestion
	if(ishumanbasic(M))
```

https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/is_helpers.dm#L69

File: code/modules/clothing/neck/_neck.dm [hunk 3]
```diff
@@ -180,13 +180,19 @@
...
+		var/mob/living/carbon/human/H = M
+		if(H.dna?.species?.id == "human")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/neck/_neck.dm]: remove 92&3

File: code/modules/clothing/neck/_neck.dm (hunk 2 above)
Comment (Qustinnus) [reply code/modules/clothing/neck/_neck.dm]: no

File: code/modules/clothing/neck/_neck.dm (hunk 3 above)
Comment (obscolene) [reply code/modules/clothing/neck/_neck.dm]: Am I misinterpreting ShizCalev's review or is it merely incorrect?

Comment (ExcessiveUseOfCobblestone) [reply code/modules/clothing/neck/_neck.dm]: He probably forgot we already had species helpers.

Comment (obscolene) [reply code/modules/clothing/neck/_neck.dm]: species helpies

File: code/modules/clothing/neck/_neck.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/clothing/neck/_neck.dm]: It's the same thing as is currently but with the proper define.

I dunno why lizards and the like can still wear it but lol whatever.

**Comment by fludd12** [comment]
you okay there...?

**Comment by ExcessiveUseOfCobblestone** [comment]
Catgirl speak is probably the most difficult language to understand.

**Comment by obscolene** [comment]
i'[m having a strokies

**Comment by ExcessiveUseOfCobblestone** [comment]
So as of right now it's not the same as Shiz's review until you do the top one.

Current: All human mobs (regardless of species) cannot wear the collar
With my suggestion / Shiz's Review: Only pure human species cannot wear the collar

Top one makes more sense but that's between you and maintainers.

**Comment by obscolene** [comment]
I don't know why only nonhuman humans would be able to equip it. If you want me to change the PR title to specify "humanoids." I could do that.

**Comment by ExcessiveUseOfCobblestone** [comment]
I disagree with shiz's review but I was giving you the proper way to do it and wanted you to know what you were putting in lol.

**Comment by obscolene** [comment]
ShizCalev's review does not allow nonhuman humans to wear the collar. I tested it and it is functionally identical to the original PR, which is why I was confused.

**Comment by ExcessiveUseOfCobblestone** [comment]
that would not be correct lol.

**Comment by CitrusGender** [comment]
dog bless

**Pull request merged**